### ISSUE TITLE
Cannot add a link card for certain links, link card isn't loading 

### ISSUE THREAD

**Issue by hildwin** [bluesky-social/social-app#2734]
**Describe the bug**
I try to include a link to the website of the German parliament in a new post, and generate a link card, but it fails. 
I see the message "Please wait for your link card to finish loading", posting is not possible.
Even after waiting a while, I can only cancel. A link card cannot be generated. 

**To Reproduce**

1. Paste the following link into the compose box
https://www.bundestag.de/mediathek/live
2. click add link card

**Expected behavior**
Expected to have a link card generated. 

**Screenshots**
![Screenshot 2024-02-02 103032_waitloading](https://github.com/bluesky-social/social-app/assets/5394190/70089d24-15c6-4664-b58a-b2ed5984d028)

**Details**

 - Platform: Desktop Chrome Version 121.0.6167.139 Windows 10
 - App version: 1.67.0 

**Additional context**

**Comment by surfdude29** [comment]
I wonder, is the behaviour different on the app to the web? I just tried to reproduce it on iOS (1.66) and no image loaded, but I was able to make the post fine with the link card:

https://bsky.app/profile/surfdude29.ispost.ing/post/3kkh5o67bhs2z

**Comment by hildwin** [comment]
> I wonder, is the behaviour different on the app to the web? I just tried to reproduce it on iOS (1.66) and no image loaded, but I was able to make the post fine with the link card:
> 
> https://bsky.app/profile/surfdude29.ispost.ing/post/3kkh5o67bhs2z

I tried to generate a link card on Android (1.66): No image loaded, but I was able to post with link card, too.

**Comment by hildwin** [comment]
Checking the link with one of thousand OG-preview-site 
shows no error. What is the difference with bsky? E. g. https://www.opengraph.xyz/url/https%3A%2F%2Fwww.bundestag.de%2Fmediathek%2Flive

**Comment by kainsavage** [comment]
I am also experiencing this issue with the web client. I tried OP's post and it seems to be working now, but I can request a card for https://blog.teamclerks.net/post/renaissance-man/ and it will never get the image.

I checked documentation and source, but I couldn't find anything suggesting limits to size, aspect ratio, or anything else. Also, I've used cards with webp before that do work, so I know it's not a mime type issue.

I can even see the request in the network tab is trying to get the correct image: 
![image](https://github.com/bluesky-social/social-app/assets/3082056/80f931a2-dba1-4122-8f77-33652244636f)
(here's that url for copy/pasting purposes: https://cardyb.bsky.app/v1/image?url=https://blog-teamclerks-net-images.s3.amazonaws.com/500a4317-3052-4dbb-9690-de35baf9943b.webp)
but even though it's a 200 response code, the response itself is a 0x0 webp image: 
![image](https://github.com/bluesky-social/social-app/assets/3082056/d4b30896-c2f1-4c00-80fe-d59e388b2826)

**Comment by surfdude29** [comment]
Whoa that is very trippy indeed! Maybe cardyb is having trouble with the webp to png conversion? @Jacob2161 might know what's going on

**Comment by kainsavage** [comment]
Definitely something strange going on. For fun I changed the format to png (since you mentioned it) and tried to pull the card again, but it failed.
![image](https://github.com/bluesky-social/social-app/assets/3082056/145c9021-ee6e-43b1-9fcd-65900ec9d107)
![image](https://github.com/bluesky-social/social-app/assets/3082056/00ef5a71-e98d-4972-9740-668956b46582)

Accessing the [CardyB URL](https://cardyb.bsky.app/v1/image?url=https://blog-teamclerks-net-images.s3.amazonaws.com/57989664-aab8-4940-8443-ba6c7ea2406a.png) directly shows it start loading the image then quit and show a white screen with the url instead (it quits at around where it quits in that thumb from the networking tab above).

So, I don't think it's necessarily in a format conversion step... something weird about this image specifically is causing issues (and probably other images as well).

**Comment by surfdude29** [comment]
> Accessing the [CardyB URL](https://cardyb.bsky.app/v1/image?url=https://blog-teamclerks-net-images.s3.amazonaws.com/57989664-aab8-4940-8443-ba6c7ea2406a.png) directly shows it start loading the image then quit and show a white screen with the url instead (it quits at around where it quits in that thumb from the networking tab above).

Yeah when I download the png I get a very similar image where it just stops decoding/runs into errors part way down:

![IMG_2455](https://github.com/bluesky-social/social-app/assets/149612116/8bf4cb27-cac8-423b-8fb6-f30b8d862d0a)

> So, I don't think it's necessarily in a format conversion step... something weird about this image specifically is causing issues (and probably other images as well).

Yeah hopefully Jake can shed some light on what's causing the glitches 🤞

> EDIT: Also - dang! Look at the difference in file size between webp and png... bananas.

Quite a difference!

**Comment by scottleechua** [comment]
I wonder if a `<script>` element coming before the `<meta property="og:image">` element trips up cardyb. That seems to be the commonality across our examples.

On @hildwin's sample article:
```
<html xml:lang="de" lang="de" dir="ltr">
<head>
  <meta charset="utf-8">
  ...
  <script>
       // <![CDATA[\n
  </script>
  ...
  <meta property="og:image" content=...>
```

On @kainsavage 's `blog.teamclerks.net` article:
```
<html lang="en">
<head>
  <script src="/_next/static/chunks/fd9d1056-ff203e689d2aa2d9.js" async="">
  ...
  <meta property="og:image" content=...>
```

On a [news article](https://news.abs-cbn.com/news/2024/9/1/wrap-from-bamban-to-batam-alice-guo-s-accomplices-reveal-pogo-s-wide-web-848) that cardyb also failed to get an image for:
```
<html>
<head>
  <meta charSet="utf-8"/>
  ...
  <script type="text/javascript" src=...>
  ...
  <meta property="og:image" content=...>  
```

Might there be a way to peek into cardyb's HTML parsing rules? 🤔
